### ISSUE TITLE
Refreshing page after upload is done triggers unknown infoHash error in logs.

### ISSUE THREAD

**Issue by ilfarpro** [Chocobozzz/PeerTube#6516]
### Describe the current behavior

Recently I uploaded massive amount of videos and not clicked publish button after upload is done. Because Peertube lags and hang up when click this button immedieately after upload is done. 

Then I realized this user behaviour triggers "unknown infoHash" error in server logs. This is critical, because when infoHash is unknown it means no P2P for this videos.

To clarify things I did controlled test. I uploaded the same video 2 times:
1. Immediately refreshed page after upload is done, not clicked Publish button, maked video Public after transcoding. This video gets "Unknown infoHash" error when whatching.
2. Uploaded video, clicked Publish button, waited while video page opens (takes long time, sometimes not opens at all). This video plays just fine, **no** "Unknown infoHash" error, **no** temp bans from tracker (no websocket connection related errors). 

### Steps to reproduce

1. Upload video, wait till you see message that video is uploaded (shows up right after video upload is done)
2. Refresh Video, then wait when it's trancoding is done.
3. Edit video, make Public.
4. Whatch video, open browser console, you will see Websocket connection error after a while.
5. Check server logs, you will see "Unknown infoHash" error.

---
Sample data:
1. Whatch this video first https://k-pop.22x22.ru/w/a6e3QYzAmzKJyjdun5UxUj - it will not trigger infoHash error, and you will Not see websocket connection issues in browser console
2. Close that video, open another one https://k-pop.22x22.ru/w/dmJskEXujk6xpWJaKJ4Fp6 - whatch this one 3 times, you will see websocket connection error in browsers console

### Describe the expected behavior

If I upload video and not click Publish button then leave page it should not trigger unknown infoHash error.

### Additional information

* PeerTube instance:
  * URL: https://k-pop.22x22.ru/
  * Version: 6.2
  * NodeJS version: v18.20.3
  * Ffmpeg version: 7.0.1

* Browser name, version and platforms on which you could reproduce the bug: Chrome 126
* Client browser console log: 
```
WebSocket connection to 'wss:// k-pop.22x22.ru/tracker/socket ' failed
```
* Server log:
```
warn[28.07.2024, 21:29:43] Warning in tracker.
{
  "err": {
    "stack": "Error: Unknown infoHash 60c67648ba46f19f436f871a06faa1d05ccbd5d7 requested by ip xxx.xx.xxx.xxx\n    at Server.filter [as _filter] (file:///app/dist/core/controllers/tracker.js:50:16)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)",
    "message": "Unknown infoHash 60c67648ba46f19f436f871a06faa1d05ccbd5d7 requested by ip xxx.xx.xxx.xxx"
  }
}
```

**Comment by ilfarpro** [comment]
Is there any way to fix infoHash without re-transcoding videos again? Maybe manually trigger infoHashes update?

I've uploaded too many videos and they are transcoded in high quality, it took so long, it would be such a waste of time, money and effort If I will need to reupload or re-transcode again.

**Comment by Chocobozzz** [comment]
Hi,

Can you go inside your database and paste the output of:

```sql
select * from "videoStreamingPlaylist" where id = 260;
```

?

**Comment by ilfarpro** [comment]
> Hi,
> 
> Can you go inside your database and paste the output of:
> 
> ```sql
> select * from "videoStreamingPlaylist" where id = 260;
> ```
> 
> ?

Hi, @Chocobozzz 

Here is the output:
```
 id  | type |                 playlistFilename                 | playlistUrl |                                                                                                                                                         p2pMediaLoaderInfohashes                                                                                                                                                          | p2pMediaLoaderPeerVersion |                  segmentsSha256Filename                   | segmentsSha256Url | videoId | storage |         createdAt          |         updatedAt         
-----+------+--------------------------------------------------+-------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------+-----------------------------------------------------------+-------------------+---------+---------+----------------------------+---------------------------
 260 |    1 | 972d02b5-faca-4614-a121-0303a0e6b4c8-master.m3u8 |             | {128801be5d85d108daa19f0802fffc82dd68e3cf,b21634bcc821cb67c7589bde9f4f550e5cff4907,084cd56e7029eb42996cebb9e72037a4fc9784dd,b7d3c5132dbd2b68296904f8de7ab8f88c79c52b,1cecbac8742655a296f479c72ff2ad199b4ae0a7,f4f5dbf03901eb8373affb7d9de49b2bfe217590,0ac085374633ad077a4e170df6d8f9bf0c362cb7,2df416b32fa2d150c74297cac6bc82e3b3c4d49c} |                         2 | 6fee410a-373d-4f28-8dc8-12455d721a62-segments-sha256.json |                   |     133 |       0 | 2024-07-28 18:10:41.473+00 | 2024-07-28 18:14:29.37+00
```

**Comment by Chocobozzz** [comment]
Hi and thanks a lot! It should be fixed by https://github.com/Chocobozzz/PeerTube/commit/22e9d9a1fe52b4358bdebffe99d5eafb17a93e5d

Update video privacy from `public` to `private` and then to `private` to `public` should fix existing videos with 6.2.1 release.
